tests/common/mod.rs

```diff
@@ -12,6 +12,17 @@ fn test_zero() {
     getrandom_impl(&mut [0u8; 0]).unwrap();
 }
 
+// Return the number of bits in which s1 and s2 differ
+fn num_diff_bits(s1: &[u8], s2: &[u8]) -> usize {
+    assert_eq!(s1.len(), s2.len());
+    let mut n = 0;
+    for i in 0..s1.len() {
+        n += (s1[i] ^ s2[i]).count_ones() as usize;
+    }
+    n
```

Comment:
You could use a fancier approach here:
```rust
pub fn num_diff_bits(s1: &[u8], s2: &[u8]) -> u32 {
    assert_eq!(s1.len(), s2.len());
    s1.iter()
        .zip(s2.iter())
        .map(|(a, b)| (a ^ b).count_ones())
        .sum()
}

Reply:
Nice, I'm going to keep it as returning `usize` though to reduce the casting boilerplate in the tests.